src/config.rs

```diff
@@ -1,19 +1,42 @@
+use std::env;
 use crate::{Error, Result};
 use serde::{Deserialize, Serialize};
 
+fn default_terminal() -> std::string::String{
+    match env::var("TERMINAL") {
+        Ok(val) => val,
+        Err(_) => "xterm -e".to_owned()
+    }
+}
+
+fn default_selector() -> std::string::String{
+    "rofi -dmenu -p 'Open With: '".to_owned()
+}
+
+fn default_enable_selector() -> bool {
+    false
+}
+
 #[derive(Serialize, Deserialize)]
 pub struct Config {
+    #[serde(default="default_enable_selector")]
     pub enable_selector: bool,
+
+    #[serde(default="default_selector")]
     pub selector: String,
+
+    #[serde(default="default_terminal")]
     pub terminal_emulator: String,
 }
 
+
 impl Default for Config {
     fn default() -> Self {
+        // This seems repetitive but serde does not uses Default
         Config {
-            enable_selector: false,
-            selector: "rofi -dmenu -p 'Open With: '".to_owned(),
-            terminal_emulator: "xterm -e".to_owned(),
+            enable_selector: default_enable_selector(),
+            selector: default_selector(),
+            terminal_emulator: default_terminal()
```

Comment:
Hey thanks for the PR!

I think we could just get away with inlining ` default_terminal()` into `impl Default`, there's no need to create a function for each option. Unless there's something I'm missing ?

Reply:
I move the defaults to use serde annotations too because when some fields are missing and not the complete conf file, the `Default` trait is not used to get those values.